libmamba/include/mamba/core/transaction.hpp

```diff
@@ -165,6 +165,8 @@ namespace mamba
 
         History::UserRequest m_history_entry;
         Transaction* m_transaction;
+        ::Id m_real_repo_key = 0;  // Must match that in ``MRepo``
+        ::Id m_mrepo_key = 0;      // Must match that in ``MRepo``
```

Comment:
I guess this leaking of libsolv details in the transaction class is temporary and will be replaced with PackageInfo list on the long run? In the meantime, they (and those in MRepo) should call the same initialization function to ensure their "synchronization"

Reply:
This constraint was already here, the PR just made it more obvious.
At the libsolv level (which will be private), #2401 will introduce clean getter/setter in the bindings so that we don't need to match the keys manually.
At the user level, this will be invisible. The user will only deal in PackageInfo style of data.